tests/cupy_tests/random_tests/test_generator.py

```diff
@@ -253,6 +253,96 @@ def test_goodness_of_fit_2(self):
         self.assertTrue(hypothesis.chi_square_test(counts, expected))
 
 
+@testing.parameterize(
+    {'a': 3, 'size': 2, 'p': None},
+    {'a': 3, 'size': 2, 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (5, 5), 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (), 'p': None},
+    {'a': [1, 2, 3], 'size': 2, 'p': [0.3, 0.3, 0.4]},
```

Comment:
Could you add a test case where `a` is an one-dimentional array whose `dtype` is different from integer.